include/common/Driver.hpp

```diff
@@ -524,9 +551,79 @@ class Driver
 			double const* const lastYdot = _sim->getLastSolutionDerivative(len);
 
 			writer.vector("LAST_STATE_Y", len, lastY);
-			writer.vector("LAST_STATE_YDOT", len, lastYdot);
+			writer.vector("LAST_STATE_YDOT", len, lastYdot); 
 		}
 
+		/*Implementatoin of writing last state Y and YDot for each unit operation*/
+
+		for (int i = 0; i <= _sim->model()->maxUnitOperationId(); ++i)
```

Comment:
In the new approach (see below), we only need to iterate over the units in the list. Note that the unit operation id does not necessarily coincide with the index in `_models`.